lib/fs.js

```diff
@@ -41,6 +41,25 @@ const isWindows = process.platform === 'win32';
 const DEBUG = process.env.NODE_DEBUG && /fs/.test(process.env.NODE_DEBUG);
 const errnoException = util._errnoException;
 
+function getOptions(options, defaultOptions) {
+  let actualType = typeof options;
```

Comment:
FWIW v8 still does not (IIRC) optimize storing `typeof foo` in a variable. It's faster to just inline the `typeof` without a variable (e.g. `typeof actualType === 'function'`).